src/Service.Tests/Authorization/AuthorizationResolverUnitTests.cs

```diff
@@ -1204,12 +1190,167 @@ public void ValidateClientRoleHeaderClaimIsAddedToClaimsInRequestContext()
+                { "groups", "src1" },
+                { "_claim_sources", distributedClaimSources},
+                { "wids", new List<string>() { "d74b8d81-39eb-4201-bd9f-9f1c4011e3c9", "18d14519-c4da-4ad4-936d-9a2de69d33cf", "9e513fc0-e8af-43b1-a6c7-949edb1967a3" } },
+                { "int_array", new List<int>() { 1, 2, 3 } },
+                { "bool_array", new List<bool>() { true, false, true } },
+                { "roles", new List<string> { "anonymous", "authenticated", TEST_ROLE } },
+                { "nullValuedClaim",  null},
+                { "iat", 1706816426 }
+            };
+
+            ClaimsPrincipal principal = await CreateClaimsPrincipal(userClaims: claimsCollection);
+
+            Mock<HttpContext> context = new();
+            context.Setup(x => x.User).Returns(principal);
+            context.Setup(x => x.Request.Headers[AuthorizationResolver.CLIENT_ROLE_HEADER]).Returns(TEST_ROLE);
+
+            // Act
+            Dictionary<string, string> claimsInRequestContext = AuthorizationResolver.GetProcessedUserClaims(context.Object);
+
+            // Assert
+            // The only role that should be present is the role set in the client role header. The claims collection
+            // configured in the 'arrange' section of this test must include that role in the list of role claims.
+            // DAB historically only sent a single role into MS SQL's session context, so changing this behavior
+            // is a breaking change.
+            Assert.AreEqual(expected: true, claimsInRequestContext.ContainsKey(AuthenticationOptions.ROLE_CLAIM_TYPE), message: "The 'roles' claim must be present.");
+            Assert.AreEqual(expected: TEST_ROLE, claimsInRequestContext[AuthenticationOptions.ROLE_CLAIM_TYPE], message: "The roles claim should have the value:" + TEST_ROLE);
+
+            // Ensure JSON arrays are reconstructed correctly aligning to the JWT token claim's value type
+            Assert.AreEqual(expected: @"[""d74b8d81-39eb-4201-bd9f-9f1c4011e3c9"",""18d14519-c4da-4ad4-936d-9a2de69d33cf"",""9e513fc0-e8af-43b1-a6c7-949edb1967a3""]", claimsInRequestContext["wids"]);
+            Assert.AreEqual(expected: "[1,2,3]", actual: claimsInRequestContext["int_array"]);
+            Assert.AreEqual(expected: "[true,false,true]", actual: claimsInRequestContext["bool_array"]);
+            Assert.AreEqual(expected: @"{""src1"":{""endpoint"":""https://graph.microsoft.com/v1.0/users/{userID}/getMemberObjects""}}", actual: claimsInRequestContext["_claim_sources"]);
+            Assert.AreEqual(expected: "1706816426", actual: claimsInRequestContext["iat"]);
+            Assert.AreEqual(expected: "", actual: claimsInRequestContext["nullValuedClaim"]);
+        }
+
+        /// <summary>
+        /// JWT token JSON payloads may not be flat and may contain nested JSON objects or arrays.
+        /// This test validates that when dotnet (authentication jwt processing code) flattens the JWT token payload,
```

Comment:
nit: this sentence `This test validates that when dotnet (authentication jwt processing code) flattens the JWT token payload,` is incomplete

Reply:
updated. 

```csharp
        /// JWT token JSON payloads may not be flat and may contain nested JSON objects or arrays.
        /// This test validates that when dotnet's JWT processing code flattens the JWT token payload
        /// into Claim objects, the AuthorizationResolver.GetProcessedUserClaims(...) method correctly
        /// returns a dictionary where key=claimType and value= scalar value or serialized JSON array.
        /// This test enforces that DAB only processes one 'roles' claim whose value matches the x-ms-api-role
        /// header, which protects against regression.
```